tools/cbmc/proofs/make_create_header_files.py

```diff
@@ -0,0 +1,144 @@
+#!/usr/bin/env python3
+#
+# Creates and reverses headerfiles required for CBMC proofs.
+#
+# Copyright (C) 2019 Amazon.com, Inc. or its affiliates.  All Rights Reserved.
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+import os
+import re
+
+CWD = os.getcwd()
+
+CREATE_HEADERS_FOR = ["lib/FreeRTOS/queue.c"]
+
+POSIX_INCLUDE = "lib/include/FreeRTOS_POSIX"
+TCP_INCLUDE = "lib/FreeRTOS-Plus-TCP"
+FreeRTOS_INCLUDE = "lib/include/private"
+
+
+FreeRTOS_HEAD = os.path.normpath(os.path.join(CWD, "..", "..", ".."))
+
+REGEX_START = re.compile(r"/\*CBMC_HEADER_START\:([\w\.]+)\*/")
+REGEX_END = re.compile(r"/\*CBMC_HEADER_END\*/")
+
+REGEX_INJECTED = re.compile(r"/\*CREATE_HEADER_FILES\*/")
+REGEX_INCLUDE = re.compile(r"#include\"([\w\.]+)\"")
+
+def get_absolute_header(header_name, file):
+    if "FreeRTOS_POSIX" in file:
+        absolut_header = os.path.normpath(
+            os.path.join(FreeRTOS_HEAD,
+                         *POSIX_INCLUDE.split("/"),
+                         header_name))
+    elif "FreeRTOS-Plus-TCP" in file:
+        absolut_header = os.path.normpath(
+            os.path.join(FreeRTOS_HEAD,
+                         *TCP_INCLUDE.split("/"),
+                         header_name))
+    else:
+        absolut_header = os.path.normpath(
+            os.path.join(FreeRTOS_HEAD,
+                         *FreeRTOS_INCLUDE.split("/"),
+                         header_name))
+    return absolut_header
+
+def create_header(file):
+    absolut_file = os.path.join(FreeRTOS_HEAD, *file.split("/"))
```

Comment:
Minor detail, but there are several occurrences of "absolut_*" in the file which can be replaced by "absolute_*".